### PR TITLE
chore(identity-keys): set up identity keys package

### LINKED ISSUE #79: Create `@walletconnect/identity-keys` pkg
## Context

- Both chat and push clients are beginning to overlap in their interactions with the keyserver regarding identity keys
- This logic should be made common to avoid duplicated code and runtime work between clients
- ChatClient currently has its own methods for:
	- `registerIdentity`: https://github.com/WalletConnect/chat-client-js/blob/main/packages/chat-client/src/controllers/engine.ts#L102
	- `resolveIdentity`: https://github.com/WalletConnect/chat-client-js/blob/main/packages/chat-client/src/controllers/engine.ts#L271
	- `unregisterIdentity`: https://github.com/WalletConnect/chat-client-js/blob/main/packages/chat-client/src/controllers/engine.ts#L377
- PushClient currently only requires the `registerIdentity` step. All push-facing work on IdentityKeys is encapsulated in [this PR](https://github.com/WalletConnect/push-client-js/pull/4/files#diff-bad14c9827b76e73706e79a2f97ac2aa4e58afe689b4fad473fd2f806ceb79b1R810)

## Implementation

- Create a new package for `@walletconnect/identity-keys` in `misc`
- Hoist Identity keys functionality from Chat and Push Clients into the `IdentityKeys` class. Let clients pass `core` into the class as part of the constructor.
- Inside the `IdentityKeys` class, set up a `Store` instance dedicated solely to storing/persisting identity keys, similar to [how this already works in PushWalletClient](https://github.com/WalletConnect/push-client-js/blob/main/packages/push-client/src/walletClient.ts#L77), but with the accounting for the keys abstracted into the `IdentityKeys` package under the guaranteed same storage key, rather than having each client run its own store like currently.
- Release a canary preview and dogfood abstracted flow in both Chat and PushClients to ensure feature parity

## Proposed API

```ts
public abstract registerIdentity: (params: {
    accountId: string,
    onSign: (message: string) => Promise<string>
	keyserverUrl?: string;
  }): Promise<string>;

public abstract resolveIdentity(params: {
    publicKey: string;
	keyserverUrl?: string;
  }): Promise<Cacao>;

public abstract unregisterIdentity(params: {
    account: string;
	identityKeyPub: string;
	keyserverUrl?: string;
  }): Promise<void>;
```

### PR DESCRIPTION
**Pull request by Cali93**
Closes #79

### CODE REVIEW

File: misc/identity-keys/src/identity-keys.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        },
+      };
+
+      const cacaoMessage = formatMessage(cacao.p, composeDidPkh(accountId));
+
+      const signature = await onSign(cacaoMessage);
+
+      // Storing keys after signature creation to prevent having false statement
+      // Eg, onSign failing / never resolving but having identity keys stored.
+      this.identityKeys.set(`${accountId}_identityKeys`, {
```
Comment (bkrem) [inline misc/identity-keys/src/identity-keys.ts:90]: I think having the `_identityKeys` suffix on the key is redundant now that it's not a generic store but a concrete one owned by the `IdentityKeys` class. Using simply `accountId: string` as the index (as it is currently in the client implementations) should be sufficient now.

```suggestion
      this.identityKeys.set(accountId, {
```

**Review by bkrem — APPROVED** [review]

File: misc/identity-keys/src/identity-keys.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+const DEFAULT_KEYSERVER_URL = "https://keys.walletconnect.com/";
+const IDENTITY_KEYS_STORAGE_PREFIX = "wc@2:identityKeys";
+
+export class IdentityKeys implements IIdentityKeys {
+  private keyserverUrl: string;
+  private identityKeys: IStore<IdentityKeychain["accountId"], IdentityKeychain>;
+
+  constructor(private core: ICore, keyServerUrl?: string) {
+    this.keyserverUrl = keyServerUrl ?? DEFAULT_KEYSERVER_URL;
+    this.identityKeys = new Store(
```
Comment (devceline) [inline misc/identity-keys/src/identity-keys.ts:31]: Having a separate st

File: package.json [hunk 3]
```diff
@@ -31,5 +31,8 @@
     "lerna": "^6.5.1",
     "npm-run-all": "^4.1.5",
     "tsc": "^2.0.3"
+  },
+  "dependencies": {
+    "@walletconnect/did-jwt": "^2.0.0"
```
Comment (devceline) [inline package.json:36]: Hope I'm not missing something but why are we adding a dependency at the root level?

**Review by devceline — APPROVED** [review]
LGTM barring one question about dep management

File: misc/identity-keys/src/identity-keys.ts (hunk 2 above)
Comment (bkrem) [reply misc/identity-keys/src/identity-keys.ts:31]: While you were OOO, Cali and I realised that what we discussed initially created more problems than it solved. Something we didn’t properly account for in our original discussion was that our `Store` interface itself consumes the core (all existing stores live at client level not core level).

We decided that it’s simpler and keeps modularity to treat this class the same way and let it manage a store, removing any additional burden and complexity on the core controller.

You can see the awkward situation Cali ran into here:
https://github.com/WalletConnect/walletconnect-monorepo/pull/2190

File: package.json (hunk 3 above)
Comment (bkrem) [reply package.json:36]: Think it’s accidental 👍

cc @Cali93

File: misc/identity-keys/src/identity-keys.ts (hunk 2 above)
Comment (devceline) [reply misc/identity-keys/src/identity-keys.ts:31]: Details like that do come up in implementation, thank you for clearing it up

File: package.json [hunk 4]
```diff
@@ -31,5 +31,9 @@
     "lerna": "^6.5.1",
     "npm-run-all": "^4.1.5",
     "tsc": "^2.0.3"
-  }
+  },
+  "optionalDependencies": {
+    "@nrwl/nx-linux-x64-gnu": "^15.9.2"
+  },
+  "version": "0.1.0-3106dd9"
```
Comment (bkrem) [inline package.json:38]:
```suggestion

```

**Pull request merged**